src/System.Buffers/src/System/Buffers/NativeBufferPool.cs

```diff
@@ -8,12 +8,31 @@
 
 namespace System.Buffers
 {
-    public unsafe sealed class NativeBufferPool<T> where T : struct
+    public unsafe sealed class NativeBufferPool<T> : IDisposable where T : struct
     {
         private const int NUMBER_OF_BUFFERS_IN_BUCKET = 50;
         private int _maxElements;
         private NativeBufferBucket<T>[] _buckets;
 
+        private static NativeBufferPool<byte> _sharedInstance = null;
+
+        public static NativeBufferPool<byte> SharedByteBufferPool
+        {
```

Comment:
I'll argue that allocating 2MB of unnecessary memory is a heavier hammer.

Allocations are not free nor cheap. I firmly believe the mutex (use `Monitor` here for performance) is cheaper.